crates/lsp-ai/src/memory_backends/vector_store.rs

```diff
@@ -0,0 +1,1062 @@
+                    },
+                    position: Position {
+                        line: 0,
+                        character: 11,
+                    },
+                },
+                PromptType::ContextAndCode,
+                &json!({}),
+            )
+            .await?;
+        let prompt: ContextAndCodePrompt = prompt.try_into()?;
+        assert_eq!(prompt.context, "--file:///filler2.py--\nprint('test')");
+        assert_eq!(prompt.code, "--file:///filler.py--\n# Multiplie");
+        // Test a FIM prompt
+        let prompt = vector_store
+            .build_prompt(
+                &TextDocumentPositionParams {
+                    text_document: TextDocumentIdentifier {
+                        uri: text_document1.uri.clone(),
+                    },
+                    position: Position {
+                        line: 0,
+                        character: 10,
+                    },
+                },
+                PromptType::FIM,
+                &json!({}),
+            )
+            .await?;
+        let prompt: FIMPrompt = prompt.try_into()?;
+        assert_eq!(
+            prompt.prompt,
+            "--file:///filler2.py--\nprint('test')\n\n# Multipli"
+        );
+        assert_eq!(
+            prompt.suffix,
+            "es two numbers\ndef multiply_two_numbers(x, y):\n    return\n\n# A singular test\nassert multiply_two_numbers(2, 3) == 6\n"
+        );
+        Ok(())
+    }
+
+    #[test]
+    #[cfg(feature = "stress_test")]
+    fn stress_test_f32() -> anyhow::Result<()> {
+        let mut vector_store = VS::new(VectorDataType::F32);
+        let embedding: Vec<f32> = (0..1024).map(|x| x as f32).collect();
+        // Time insert
+        // Insert 100_000 files each with 10 chunks
+        let now = std::time::Instant::now();
+        for i in 0..100_000 {
+            let uri = format!("file://test{i}.py");
+            let mut chunks = vec![];
+            for ii in 0..10 {
+                let mut eb = embedding.clone();
+                eb[0] = i as f32;
+                eb[1] = ii as f32;
+                let stored_chunk = StoredChunk::new(
+                    uri.clone(),
+                    StoredChunkVec::new(VectorDataType::F32, eb.clone()),
+                    format!("abc-{i}-{ii}"),
+                    ByteRange::new(0, 0), // This is wrong but its ok
+                );
+                chunks.push(stored_chunk);
+            }
+            vector_store.store.insert(uri.clone(), chunks);
+        }
+        let elapsed_time = now.elapsed();
+        println!("Insert took {} milliseconds.", elapsed_time.as_millis());
+        // Time search
+        let now = std::time::Instant::now();
+        vector_store.search(5, None, embedding, "", 0)?;
+        let elapsed_time = now.elapsed();
+        println!("Search took {} milliseconds.", elapsed_time.as_millis());
+        Ok(())
+    }
+
+    #[test]
+    #[cfg(feature = "stress_test")]
```

Comment:
You can also create benchmarks with the criterion crate if need be

Reply:
I will look into that!

Reply:
You can check out the code I did for `tinyvec-embed` in the multi file context PR of llm-ls: https://github.com/huggingface/llm-ls/pull/51/files#diff-fcfd55b338e3b2f8e701d92a4cb8ecd6deca7f67f3f9f2c2439c79c7c2e72076

Reply:
Looks amazing. I will add it thank you!